src/sst_iter.rs

```diff
@@ -22,25 +21,173 @@ enum FetchTask {
     Finished(VecDeque<Arc<Block>>),
 }
 
-pub(crate) struct SstIterator<'a, H: AsRef<SsTableHandle> = &'a SsTableHandle> {
```

Comment:
😍 just to make sure I got it right, now we have one fewer lifetime to manage because the SstIterator just takes the same lifetime as the SsTableHandle if it's `Borrowed`?

Reply:
Yeah. It is more intuitive, right? Now the lifetime of the iterator is tied directly to the handle that it is iterating over. 
I found the original pattern awkward to work with because the lifetimes were indirect and we had to use the phantom data just to make the compiler happy.